### PR TITLE
Define __all__ for uarray

### PR DESCRIPTION
**Pull request by peterbell10**
Since `__init__.py` imports `*`, it exports things like `Callable` or `atexit`.

### CODE REVIEW

File: uarray/_backend.py [hunk 1]
```diff
@@ -27,6 +27,22 @@
...
 
+__all__ = [
+    "set_backend",
+    "set_global_backend",
+    "skip_backend",
+    "register_backend",
+    "clear_backends",
+    "create_multimethod",
+    "generate_multimethod",
+    "_Function",
```
Comment (hameerabbasi) [inline uarray/_backend.py]: Private variables really shouldn't be in here.

Comment (hameerabbasi) [reply uarray/_backend.py]: If you need to access it in tests, use `# type: ignore`

Comment (peterbell10) [reply uarray/_backend.py]: If you do `from uarray import *` then `_Function` still isn't included. However, sometimes you really do need to access `_Function` so it helps to have it in the `uarray` module itself.

Comment (hameerabbasi) [reply uarray/_backend.py]: As long as it isn't in `uarray.__all__`.

**Comment by hameerabbasi** [comment]
Feel free to push the button. 😄

**Pull request merged**